tests/lean/pp_parens.lean.expected.out

```diff
@@ -0,0 +1,8 @@
+((1 + (2 * 3)) + 4) : ℕ
+nat.add_assoc : ∀ (n m k : ℕ), (((n + m) + k) = (n + (m + k)))
+(([1, 2, 3]) ++ ([3, 4, 5])) : list ℕ
+∀ (x : ℤ), ℤ → ℤ → ((x - (-3)) = ((-2) + x)) : Prop
+∀ (p : ℕ → ℕ → Prop), p (1 + 2) 3 : Prop
+∀ (f : ℕ → ℕ → ℕ), ((f 2 (f 3 4 + 1) + 2) = f 1 2) : Prop
```

Comment:
Shouldn't there be parentheses around `f 3 4` and `f 1 2` in this example? It's not always obvious whether a notation has precedence higher than application or not.

Reply:
@digama0 Here's an example for how high-precedence notation interacts with the feature:
```lean
local infix ` ++ `:5000 := list.append

def tail {α : Type*} : list α → list α := list.tail

#check (tail [1,2,3])++[3,4,5]
/- (tail [1, 2, 3]) ++ [3, 4, 5] : list ℕ -/
#check tail [1,2,3]++[3,4,5]
/- tail [1, 2, 3] ++ [3, 4, 5] : list ℕ -/

set_option pp.parens true

#check (tail [1,2,3])++[3,4,5]
/- ((tail ([1, 2, 3])) ++ ([3, 4, 5])) : list ℕ -/
#check tail [1,2,3]++[3,4,5]
/- tail (([1, 2, 3]) ++ ([3, 4, 5])) : list ℕ -/
```
It seems like everything's unambiguous, even if you don't know any of the precedences. (Or, at least, if you assume function application has highest precedence.)

I had looked a little into having function applications *always* be wrapped in an additional layer of parentheses, but at least in the few easy-ish ways I'd thought of implementing it, you'd get things like `(((f) 3) 4)` or `((f x y) + (x))`.